e2e/inspect/inspect.go

```diff
@@ -0,0 +1,215 @@
+			"90-environment.sh": "#!/bin/sh\n#Custom environment shell code should follow\n\n\nexport test=\"testing\"\nexport e2e=\"e2e testing\"\n\n\n"
+		}
+	},
+	"type": "container"
+}
+`
+
+func runInspectCommand(inspectType string) ([]byte, error) {
+	argv := []string{"inspect", "--json", inspectType, containerTesterSIF}
+	cmd := exec.Command(testenv.CmdPath, argv...)
+
+	return cmd.CombinedOutput()
+}
+
+func singularityInspect(t *testing.T) {
+	tests := []struct {
+		name      string
+		insType   string   // insType the type of 'inspect' flag, eg. '--deffile'
+		json      []string // json is the path to a value that we will test
+		expectOut string   // expectOut should be a string of expected output
+	}{
+		{
+			name:      "label maintainer",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "MAINTAINER"},
+			expectOut: expectedLabelsJSON,
+		},
+		{
+			name:      "label",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "E2E"},
+			expectOut: expectedLabelsJSON,
+		},
+		{
+			name:      "label",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "HI"},
+			expectOut: expectedLabelsJSON,
+		},
+		{
+			name:      "label",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "e2e"},
+			expectOut: expectedLabelsJSON,
+		},
+		{
+			name:      "label",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "hi"},
+			expectOut: expectedLabelsJSON,
+		},
+		{
+			name:      "label",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "org.label-schema.usage"},
+			expectOut: expectedLabelsJSON,
+		},
+		{
+			name:      "label",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "org.label-schema.usage.singularity.deffile.bootstrap"},
+			expectOut: expectedLabelsJSON,
+		},
+		{
+			name:      "label",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "org.label-schema.usage.singularity.deffile.from"},
+			expectOut: expectedLabelsJSON,
+		},
+		{
+			name:      "label",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "org.label-schema.usage.singularity.runscript.help"},
+			expectOut: expectedLabelsJSON,
+		},
+		{
+			name:      "runscript",
+			insType:   "--runscript",
+			json:      []string{"attributes", "runscript"},
+			expectOut: expectedRunscriptJSON,
+		},
+		{
+			name:      "list apps",
+			insType:   "--list-apps",
+			json:      []string{"attributes", "apps"},
+			expectOut: expectedListAppsJSON,
+		},
+		{
+			name:      "test",
+			insType:   "--test",
+			json:      []string{"attributes", "test"},
+			expectOut: expectedTestJSON,
+		},
+		{
+			name:      "environment",
+			insType:   "--environment",
+			json:      []string{"attributes", "environment", "90-environment.sh"},
+			expectOut: expectedEnvironmentJSON,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, test.WithPrivilege(func(t *testing.T) {
```

Comment:
If I understand correctly, this test is supposed to only run in privileged mode so I would suggest to add `test.EnsurePrivilege(t)` at the beginning of the test rather than escalating privilege for each sub-test

Reply:
Actually it should probably be `test.WithoutPrivilege`. Not sure how it changed :confused: